compiler/src/dotty/tools/dotc/transform/SyntheticMembers.scala

```diff
@@ -525,7 +525,7 @@ class SyntheticMembers(thisPhase: DenotTransformer) {
           val pat = Typed(untpd.Ident(nme.WILDCARD).withType(patType), TypeTree(patType))
           CaseDef(pat, EmptyTree, Literal(Constant(idx)))
         }
-      Match(param, cases)
+      Match(param.annotated(New(ref(defn.UncheckedAnnot.typeRef))), cases)
```

Comment:
`New` nodes should not appear without also calling the constructor, so you can use this helper method that does that for you (just apply `New` with an additional argument list)
```suggestion
      Match(param.annotated(New(ref(defn.UncheckedAnnot.typeRef), Nil)), cases)
```

Reply:
So should all the `.annotated(New(ref(defn.UncheckedAnnot.typeRef)))` calls already in the codebase also change?

Reply:
Oh I see, they should be changed also, we have issue https://github.com/lampepfl/dotty/issues/10113 to track this